src/client/datascience/jupyter/kernels/types.ts

```diff
@@ -44,7 +44,7 @@ export type LiveKernelConnectionMetadata = Readonly<{
  * This could be a raw kernel (spec might have path to executable for .NET or the like).
  * If the executable is not defined in kernelspec json, & it is a Python kernel, then we'll use the provided python interpreter.
  */
-export type KernelSpecConnectionMetadata = Readonly<{
+export type LocalKernelSpecConnectionMetadata = Readonly<{
```

Comment:
Do you think we could get rid of the 'startUsingPythonInterpreter' type and just use the existence of the interpreter in the LocalKernelSpecConnectionMetadata instead?

Reply:
I persoanlly find the current approach much cleaner, i.e. looking for the name is better than checkinf if `interpreeter` exists or not.
Also that doesn't work for non pytohn kernels. 
E.g. if you have a Java kernel, the interpreter could exist, this means we need to actiavte the interpreter and then run the java kernel.
E.g. conda that registers a java kernel.
Sure we could check the language and make that determination, however I personally find the explicit name to be better (i.e. explicit), instead of checkinf multiple flags.

If we removed this then all of the code where we have `type === startUsingPythonInterpreter` would change to `interpreer !== undefined && kernelspec.langauge === python`
Which feels clunky, miss one thing on some place, you're stuffed.

Reply:
Maybe a different way to do this would be change ```LocalKernelSpecConnectionMetadata``` to ```NonPythonKernelSpecConnectionMetadata```?

Reply:
I find it confusing we have 'start with interpreter' and 'kernel spec with interpreter maybe'.